app/controllers/pieces_controller.rb

```diff
@@ -0,0 +1,45 @@
+class PiecesController < ApplicationController
+  def select
+    @game = Game.find(params[:id])
+    @pieces = Piece.find(params[:piece_id])
+    @pieces = @game.pieces
+    @piece_id = params[:piece_id]
+    @x_coord = params[:x_coord]
+    @y_coord = params[:y_coord]
+  end
+
+  def update
+  	@pieces = Piece.find(params[:piece_id])
+  	@game = @pieces.game
+  	@pieces.update_attributes piece_params
```

Comment:
Should `valid_move?` and `move_to!` be called before up `update_attributes`? I think yes, but am unfamiliar with your strategy.